asset/asset.go

```diff
@@ -967,6 +968,27 @@ func (a *Asset) HasGenesisWitness() bool {
 	return *witness.PrevID == ZeroPrevID
 }
 
+// HasGenesisWitnessForGroup determines whether an asset has a valid witness
+// for a genesis asset in an asset group. This asset must have a non-empty group
+// key, and a single prevWitness with a zero PrevID, empty split commitment
+// proof, and non-empty witness.
+func (a *Asset) HasGenesisWitnessForGroup() bool {
+	if a.GroupKey == nil || len(a.PrevWitnesses) != 1 {
```

Comment:
I think this still needs some change, we actually have two new cases:

-An asset with the existing genesis asset witness, that also has the group key set (needs a witness generated).
-An grouped genesis asset with a witness (empty PrevID, nil SplitCommitment, non-nil TxWitness, non-nil GroupKey), should still be considered a genesis asset

Reply:
This has been addressed with two new methods, `NeedsGenesisWitnessForGroup` and `HasGenesisWitnessForGroup`.

The `Needs` variant covers an asset with a genesis asset witness, and a group key set. This is used as a sanity check within the `GenesisTxBuilder` implementation, and to branch on in the virtual TX construction logic.

The `Has` variant is used in the VM and the proof verification logic, since at those points any genesis grouped asset should have its group witness attached, and now it only needs to be validated. At those points an asset with an empty group witness would be considered invalid.